avocado/utils/process.py

```diff
@@ -269,17 +269,34 @@ class CmdResult(object):
     :type duration: float
     :param pid: ID of the process
     :type pid: int
+    :param encoding: the encoding to use for the text version
+                     of stdout and stderr
+    :type encoding: str
     """
 
     def __init__(self, command="", stdout="", stderr="",
-                 exit_status=None, duration=0, pid=None):
+                 exit_status=None, duration=0, pid=None,
+                 encoding=None):
         self.command = command
         self.exit_status = exit_status
+        #: The raw stdout (bytes)
         self.stdout = stdout
+        #: The raw stderr (bytes)
         self.stderr = stderr
         self.duration = duration
         self.interrupted = False
         self.pid = pid
+        if encoding is None:
+            encoding = os.environ.get("PYTHONENCODING", "utf-8")
```

Comment:
shouldn't we try `sys.getdefaultencoding()` first, at least in Python3, beforing setting `utf-8` as the default?

Reply:
yep, I'd recommend `sys.getdefaultencoding()` or fallback to `PYTHONENCODING` when it's `ascii`. It should work on both and systems that really use only ASCII should not produce the extended chars, so ascii+utf-8 should not break (that much).

Reply:
Btw I thing we should create a `avocado.utils.encoding` util with method to get system encoding in a deterministic way...

Reply:
Actually, I think this approach I proposed is broken.  I was influenced by https://github.com/avocado-framework/aexpect/commit/5aa7d475ca5f095057485ee61f98d989ee41872c but looking at this approach closer, I can't find any documentation on `PYTHONENCODING` on Python docs.

There's `PYTHONIOENCODING` ( https://docs.python.org/2/using/cmdline.html#envvar-PYTHONIOENCODING ) but its purpose is limited to stdin/stdout/stderr.

I'm closing this PR and will propose something based on better documented practices (such as `sys.getdefaultencoding()`) that you guys suggested here.